### ISSUE TITLE
CheckoutConfirmTemplateSubscriber breaks when accessing issuer code

### ISSUE THREAD

**Issue by janmartenjongerius** [MultiSafepay/shopware6#6]
Shopware version: `6.4.6.0 stable`
Plugin version: `2.5.0`

It appears an event subscriber tries to access the private property `MultiSafepay\Api\Issuers\Issuer::$code` in an external context.
Should use public method `\MultiSafepay\Api\Issuers\Issuer::getCode` instead.

See: `src/Subscriber/CheckoutConfirmTemplateSubscriber.php:145`

Stack trace:

```
Error: Cannot access private property MultiSafepay\Api\Issuers\Issuer::$code
#21 /vendor/store.shopware.com/mltisafemultisafepay/src/Subscriber/CheckoutConfirmTemplateSubscriber.php(145): MultiSafepay\Shopware6\Subscriber\CheckoutConfirmTemplateSubscriber::getRealIdealName
#20 /vendor/store.shopware.com/mltisafemultisafepay/src/Subscriber/CheckoutConfirmTemplateSubscriber.php(103): MultiSafepay\Shopware6\Subscriber\CheckoutConfirmTemplateSubscriber::addMultiSafepayExtension
#19 /vendor/symfony/event-dispatcher/EventDispatcher.php(270): Symfony\Component\EventDispatcher\EventDispatcher::Symfony\Component\EventDispatcher\{closure}
#18 /vendor/symfony/event-dispatcher/EventDispatcher.php(230): Symfony\Component\EventDispatcher\EventDispatcher::callListeners
#17 /vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\Component\EventDispatcher\EventDispatcher::dispatch
#16 /vendor/shopware/core/Content/Flow/Dispatching/FlowDispatcher.php(50): Shopware\Core\Content\Flow\Dispatching\FlowDispatcher::dispatch
#15 /vendor/shopware/core/Framework/Webhook/WebhookDispatcher.php(94): Shopware\Core\Framework\Webhook\WebhookDispatcher::dispatch
#14 /vendor/shopware/core/Framework/Event/BusinessEventDispatcher.php(52): Shopware\Core\Framework\Event\BusinessEventDispatcher::dispatch
#13 /vendor/shopware/core/Framework/Event/NestedEventDispatcher.php(32): Shopware\Core\Framework\Event\NestedEventDispatcher::dispatch
#12 /vendor/shopware/storefront/Page/Checkout/Confirm/CheckoutConfirmPageLoader.php(101): Shopware\Storefront\Page\Checkout\Confirm\CheckoutConfirmPageLoader::load
#11 /vendor/shopware/storefront/Controller/CheckoutController.php(135): Shopware\Storefront\Controller\CheckoutController::confirmPage
#10 /vendor/symfony/http-kernel/HttpKernel.php(156): Symfony\Component\HttpKernel\HttpKernel::handleRaw
#9 /vendor/symfony/http-kernel/HttpKernel.php(78): Symfony\Component\HttpKernel\HttpKernel::handle
#8 /vendor/symfony/http-kernel/Kernel.php(199): Symfony\Component\HttpKernel\Kernel::handle
#7 /vendor/symfony/http-kernel/HttpCache/SubRequestHandler.php(85): Symfony\Component\HttpKernel\HttpCache\SubRequestHandler::handle
#6 /vendor/symfony/http-kernel/HttpCache/HttpCache.php(479): Symfony\Component\HttpKernel\HttpCache\HttpCache::forward
#5 /vendor/symfony/http-kernel/HttpCache/HttpCache.php(452): Symfony\Component\HttpKernel\HttpCache\HttpCache::fetch
#4 /vendor/symfony/http-kernel/HttpCache/HttpCache.php(346): Symfony\Component\HttpKernel\HttpCache\HttpCache::lookup
#3 /vendor/symfony/http-kernel/HttpCache/HttpCache.php(224): Symfony\Component\HttpKernel\HttpCache\HttpCache::handle
#2 /vendor/shopware/core/HttpKernel.php(181): Shopware\Core\HttpKernel::doHandle
#1 /vendor/shopware/core/HttpKernel.php(81): Shopware\Core\HttpKernel::handle
#0 /public/index.php(83): null
```

I see that the more recent versions of this plugin aren't hosted within this repository any longer. @Jasper-MultiSafepay if you have a better location where I can report this, please point me in the right direction.

**N.B.:** It may have been fixed in `2.5.1`, but that version is constrained to ~Shopware 6.3~ Guzzle 6, which is not a solution for 6.4 projects.

**Comment by mikededecker1** [comment]
hi @johmanx10 

version `2.5.1` is be available for Shopware versions 6.3 and above, and should be available in the Shopware store for versions 6.3 and 6.4. So could you clarify what you mean with constrained so we could help you further with this?

**Comment by janmartenjongerius** [comment]
Hi @Mike-MultiSafepay my previous message is a bit confusing, as version `2.5.1` of this package does not explicitly deny Shopware 6.4 and up. However, it does so implicitly through the added dependency on the Guzzle adapter package.

Shopware 6.4 and up depend on `guzzlehttp/guzzle` versions `~7.2`. See https://github.com/shopware/platform/blob/trunk/composer.json#L70

This conflicts with the newly added dependency on `php-http/guzzle6-adapter` in version `2.5.1` of this package, which prevents me from installing that specific patch version in Shopware 6.4 projects.

**Comment by janmartenjongerius** [comment]
I see this change is the cause: https://github.com/MultiSafepay/shopware6/compare/2.5.0...2.5.1#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R15

![image](https://user-images.githubusercontent.com/642417/143241448-4621d67e-f3c9-4d10-a1c4-75a951847912.png)

You could solve that by moving it from the `require` to `require-dev` section in `composer.json` and adding a suggestion through [`suggest`](https://getcomposer.org/doc/04-schema.md#suggest).

If that is not a satisfactory solution, you probably need to write your own adapter, so as to not conflict with Guzzle 7 dependencies.

**Comment by mikededecker1** [comment]
We added a fix to our internal repository that should solve this issue. We moved from the guzzle adapter to an other `psr/http-client-implementation`. This one should not conflict with Shopware or Guzzle.

We will probably do an release somewhere these upcoming days

**Comment by janmartenjongerius** [comment]
Thanks @Mike-MultiSafepay. I'll check for the availability of the new version soon. Once that version is released, I consider this issue closed.

**Comment by mikededecker1** [comment]
We just released version 2.5.2 which should resolve this issue.

If this issue still persist please reopen the issue.
